test-cypress/integration/setup/namespaces.spec.js

```diff
@@ -0,0 +1,352 @@
+
+        cy.visit('/namespaces');
+
+        waitUntilPageIsLoaded();
+    });
+
+    afterEach(() => {
+        cy.deleteRepository(repositoryId);
+    });
+
+    function waitUntilPageIsLoaded() {
+        // Workbench loading screen should not be visible
+        cy.get('.ot-splash').should('not.be.visible');
+
+        // No active loader
+        cy.get('.ot-loader').should('not.be.visible');
+
+        getAddNamespaceForm().should('be.visible');
+    }
+
+    const DEFAULT_NAMESPACES = {
+        'gn': 'http://www.geonames.org/ontology#',
+        'owl': 'http://www.w3.org/2002/07/owl#',
+        'rdf': 'http://www.w3.org/1999/02/22-rdf-syntax-ns#',
+        'rdfs': 'http://www.w3.org/2000/01/rdf-schema#',
+        'wgs': 'http://www.w3.org/2003/01/geo/wgs84_pos#',
+        'xsd': 'http://www.w3.org/2001/XMLSchema#'
+    };
+
+    it('verify initial state', () => {
+        getNoNamespacesAlert().should('not.be.visible');
+
+        // Should be able to insert new prefix
+        getNamespacePrefixField().should('be.visible').and('not.be.disabled');
+        getNamespaceValueField().should('be.visible').and('not.be.disabled');
+        getAddNamespaceButton().should('be.visible').and('not.be.disabled');
+
+        // Should render a table with some default namespaces
+        getNamespacesTable().should('be.visible');
+        getNamespaces().should('have.length', Object.keys(DEFAULT_NAMESPACES).length);
+
+        // Should provide pagination options
+        getNamespacesPerPageMenu().within(() => {
+            // Should show all namespaces by default (they are only 6 so they can be visualized all at once)
+            cy.get('.dropdown-toggle')
+                .should('contain', 'All')
+                .click();
+            cy.get('.page-size-option')
+                .should('have.length', 1)
+                .and('contain', 'All');
+            // Close the menu to avoid overlapping other elements
+            cy.get('.dropdown-toggle').click();
+        });
+
+        // Should show summary of results
+        getNamespacesHeaderPaginationInfo()
+            .should('be.visible')
+            .and('contain', 'Showing 1 - 6 of 6 results');
+
+        // Both header & footer pagination must be the same
+        getNamespacesHeaderPagination()
+            .should('be.visible')
+            .find('li')
+            // Single page + First & Last buttons
+            .should('have.length', 3);
+        getNamespacesPagination()
+            .should('be.visible')
+            .find('li')
+            // Single page + First & Last buttons
+            .should('have.length', 3);
+
+        // Verify default namespaces are present and interactable
+        const defaultPrefixes = Object.keys(DEFAULT_NAMESPACES);
+        getNamespaces().each(($row, $index) => {
+            const expectedPrefix = defaultPrefixes[$index];
+            const expectedNamespace = DEFAULT_NAMESPACES[expectedPrefix];
+
+            cy.wrap($row)
```

Comment:
umm what about using the `root()` function?
```javascript
            cy.wrap($row)
                .find('.namespace-prefix')
                .should('contain', expectedPrefix)
                .root()
                .find('.namespaceURI')
                .should('contain', expectedNamespace)
                .root()
                .find('.select-namespace')
                .should('be.visible')
                .and('not.be.disabled')
                .root()
                .find('.edit-namespace-btn')
                .should('be.visible')
                .and('not.be.disabled')
                .root()
                .find('.delete-namespace-btn')
                .should('be.visible')
                .and('not.be.disabled');
```

Reply:
Didn't know about it but I'm not sure it will work with `each()` because the root element is not `$row`.

Reply:
It's the `cy.wrap($row)`'s outcome what's considered as root, so it should be working.

Reply:
This will work that way only if it is invoked inside `within()`